### PR TITLE
Adding Swift Examples

### PR DESCRIPTION
**Pull request by mRs-**
Like mentioned https://github.com/pinterest/PINRemoteImage/issues/115 I will adding a complete Swift example application and adding all swift style explanation to the project

### CODE REVIEW

**Comment by garrettmoon** [comment]
Awesome! Let me know when you think this is ready to be reviewed and merged, though it might be more conducive to get a Swift expert to review?

Also, can you remove the pod update changes from this PR? I'll try and update them when I merge and that way it's easier to review.

**Comment by mRs-** [comment]
A Swift expert would be nice to review this. Maybe I learn something new ;)

pod update changes will be dropped with the last commit, when I am completely ready with this here :)

**Comment by mRs-** [comment]
It's done :)

File: Swift-Example/Swift-Example/ProcessingViewController.swift [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    override func viewDidAppear(animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        imageView.pin_setImageFromURL(NSURL(string: "https://s-media-cache-ak0.pinimg.com/736x/5b/c6/c5/5bc6c5387ff6f104fd642f2b375efba3.jpg")!, processorKey: "rounded") { (result :PINRemoteImageManagerResult!, cost :UnsafeMutablePointer<UInt>) -> UIImage! in
+            if let image = result.image {
```
Comment (kerrmarin) [inline Swift-Example/Swift-Example/ProcessingViewController.swift]: It's perhaps more appropriate to use `guard` here

File: Swift-Example/Swift-Example/ProcessingViewController.swift [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+//  Copyright © 2016 Marius Landwehr. All rights reserved.
+//
+
+import UIKit
+import PINRemoteImage
+
+class ProcessingViewController: UIViewController {
+
+    @IBOutlet weak var imageView: UIImageView!
+    override func viewDidLoad() {
```
Comment (rahul-malik) [inline Swift-Example/Swift-Example/ProcessingViewController.swift]: We should remove empty method implementations if they aren't needed.

File: Swift-Example/Swift-Example/ProcessingViewController.swift (hunk 1 above)
Comment (rahul-malik) [inline Swift-Example/Swift-Example/ProcessingViewController.swift]: +1 @kerrmarin  to use `guard` here

File: Swift-Example/Swift-Example/ProcessingViewController.swift [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                UIGraphicsEndImageContext()
+                
+                return processedImage
+            } else {
+                return UIImage()
+            }
+        }
+    }
+
+    /*
```
Comment (rahul-malik) [inline Swift-Example/Swift-Example/ProcessingViewController.swift]: These are probably generated comments for new view controllers? Lets remove them.

File: Swift-Example/Swift-Example/ProgressiveViewController.swift [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    @IBOutlet weak var imageView: UIImageView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Do any additional setup after loading the view.
+        imageView.pin_updateWithProgress = true
+    }
+
+    override func didReceiveMemoryWarning() {
```
Comment (rahul-malik) [inline Swift-Example/Swift-Example/ProgressiveViewController.swift]: remove empty method implementation

File: Swift-Example/Swift-Example/ScrollViewController.swift [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+        }
+        
+        return cell
+    }
+    
+    func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return collectionKittens.count
+    }
+    
+    func numberOfSectionsInCollectionView(collectionView: UICollectionView) -> Int {
```
Comment (rahul-malik) [inline Swift-Example/Swift-Example/ScrollViewController.swift]: This method can be deleted since 1 section is the default

**Comment by garrettmoon** [comment]
@mRs- I'm sorry this never got merged in :( Now that Swift 3 is out this probably needs to be updated. I'm going to close for now, but if you update it I'll happily reopen!

**Comment by mRs-** [comment]
@garrettmoon updated it 💪

**Comment by mRs-** [comment]
Moved it to: https://github.com/pinterest/PINRemoteImage/pull/280

**Pull request closed without merging**